### DISCUSSION TITLE
Sync data to hudi task, flink report error!!please help!

### DISCUSSION THREAD

**Discussion by NiceMoon** [apache/flink-cdc#1066] [Q&A]
[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.runtime.rest.util.RestClientException: [Internal server error., <Exception on server side:
org.apache.flink.runtime.rpc.exceptions.RecipientUnreachableException: Could not send message [LocalFencedMessage(00000000000000000000000000000000, LocalRpcInvocation(JobMasterGateway.requestJob(Time)))] from sender [Actor[akka://flink/temp/jobmanager_2$r]] to recipient [Actor[akka://flink/user/rpc/jobmanager_2#1819025616]], because the recipient is unreachable. This can either mean that the recipient has been terminated or that the remote RpcService is currently not reachable.
	at org.apache.flink.runtime.rpc.akka.DeadLettersActor.handleDeadLetter(DeadLettersActor.java:61)
	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:24)
	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:20)
	at scala.PartialFunction.applyOrElse(PartialFunction.scala:123)
	at scala.PartialFunction.applyOrElse$(PartialFunction.scala:122)
	at akka.japi.pf.UnitCaseStatement.applyOrElse(CaseStatements.scala:20)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:171)

**Comment by dreamness-dnalm** [comment]
hi, I meet the same problem. Have you resolved it? Thank you.

**Comment by Thijsvandepoll** [comment]
I am also encountering this error. Any solutions?

**Reply by LiaoMinYu** [reply]
I am also encountering this error .Any solutions?

**Reply by punish-yh** [reply]
I am also encountering this error .Any solutions?

**Reply by dreamness-dnalm** [reply]
@LiaoMinYu @punish-yh I solved it by switch the version of the dependences (hudi, hadoop, hdfs etc), or you may have to recompile hudi.

**Reply by dreamness-dnalm** [reply]
I refered to the pom file of hudi offical example.

**Reply by gamblewin** [reply]
> @LiaoMinYu @punish-yh I solved it by switch the version of the dependences (hudi, hadoop, hdfs etc), or you may have to recompile hudi.

can u tell me the version of hadoop, hudi, flink you use? my environment uses hadoop-3.3.1, flink-1.16.2 and hudi-0.13.0, i change the pom file of hudi and recompile it, but still encounter this mistake.

**Reply by VietTruong-50** [reply]
I am also encountering this error .Any solutions?

**Reply by maherrt** [reply]
any updates, same issue?

**Comment by byte4tide** [comment]
I have encountered the same problem, I guess it's beacuse of the dependency conflict, My environment is flink 1.14-1.17 and hudi 1.13.X-1.14.0

**Comment by CamelliaYj** [comment]
I have encountered the same problem when I add the hive sync config，I solved this problem by adding hive-exec jar to pom. My environment is flink 1.17.0 and hudi 0.14.0 and hive 3.1.3.

**Reply by lijihuaxiaomo** [reply]
I have this version too, can you tell me how to do it?

**Reply by YunJar** [reply]
I added this
![image](https://github.com/user-attachments/assets/3da3c9cf-f61b-48f6-b0fe-fa71a37d6398)

**Comment by xiaolan-bit** [comment]
i meet the same solution with flink in kerberos
my flink sql:
CREATE CATALOG iceberg_flink_catalog WITH (
   'type'='iceberg',
   'catalog-type'='hive',
   'uri'='thrift://10.0.0.65:9083,thrift://10.0.0.139:9083',
   'clients'='5',
   'property-version'='1',
   'warehouse'='hdfs://hdfs-ha/flink/warehouse/tablespace/managed/hive'
  );

use catalog iceberg_flink_catalog;

show databases;

SET execution.runtime-mode = batch;
SET 'sql-client.execution.result-mode' = 'tableau';


select * from iceberg_spark_example.test_table;  
and when select ,there are error appear:
[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.runtime.rest.util.RestClientException: [Internal server error., <Exception on server side:
org.apache.flink.runtime.rpc.exceptions.RecipientUnreachableException: Could not send message [LocalFencedMessage(00000000000000000000000000000000, LocalRpcInvocation(JobMasterGateway.requestJob(Time)))] from sender [Actor[akka://flink/temp/jobmanager_2$e]] to recipient [Actor[akka://flink/user/rpc/jobmanager_2#879947752]], because the recipient is unreachable. This can either mean that the recipient has been terminated or that the remote RpcService is currently not reachable.
